.github/workflows/gen-release-summary.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+set -euo pipefail
+
+RELEASE_TAG="$1"
+
+# find the previous "proper" release (i.e. not VALIDATION) tag
+prev_tag="$(git tag \
+    | sort \
+    | grep -B10000 "$RELEASE_TAG" \
+    | grep -E '^[0-9]+\.[0-9]+\.[0-9]+$' \
+    | head -n -1 \
+    | tail -1)" || echo ""
+
+if [[ -z "$prev_tag" ]]; then
+    echo "Error: failed to determine previous release tag, are you sure input tag $RELEASE_TAG exists?"
+    exit 1
+else
+    echo "Previous proper release tag is: $prev_tag" >&2
```

Comment:
It will not "stand out" if you use it with redirected output, i.e. 

```
$ ./gen-release-summary.sh 2024.7.0 > foo
Previous proper release tag is: 2023.9.1
$
```

Can be removed if you feel strongly about it.